src/chocolatey/infrastructure.app/services/RegistryService.cs

```diff
@@ -290,34 +286,6 @@ private void get_msi_information(RegistryApplicationKey appKey, RegistryKey key)
                      appKey.Version = set_if_empty(appKey.Version, msiProductKey.get_value_as_string("Version"));
                      appKey.VersionMajor = set_if_empty(appKey.VersionMajor, msiProductKey.get_value_as_string("VersionMajor"));
                      appKey.VersionMinor = set_if_empty(appKey.VersionMinor, msiProductKey.get_value_as_string("VersionMinor"));
-
-                     // search components for install location if still empty
-                     // the performance of this is very bad - without this the query is sub-second
-                     // with this it takes about 15 seconds with around 200 apps installed
-                     //if (string.IsNullOrWhiteSpace(appKey.InstallLocation) && !appKey.Publisher.contains("Microsoft"))
-                     //{
-                     //    var msiComponentsKey = FaultTolerance.try_catch_with_logging_exception(
-                     //       () => msiRegistryKey.OpenSubKey("{0}\\Components".format_with(subKeyName), RegistryKeyPermissionCheck.ReadSubTree, RegistryRights.ReadKey),
-                     //       "Failed to open subkey named '{0}' for '{1}', likely due to permissions".format_with(subKeyName, msiRegistryKey.Name),
-                     //       logWarningInsteadOfError: true);
-                     //    if (msiComponentsKey == null) continue;
-
-                     //    foreach (var msiComponentKeyName in msiComponentsKey.GetSubKeyNames())
-                     //    {
-                     //        var msiComponentKey = FaultTolerance.try_catch_with_logging_exception(
-                     //           () => msiComponentsKey.OpenSubKey(msiComponentKeyName, RegistryKeyPermissionCheck.ReadSubTree, RegistryRights.ReadKey),
-                     //           "Failed to open subkey named '{0}' for '{1}', likely due to permissions".format_with(subKeyName, msiRegistryKey.Name),
-                     //           logWarningInsteadOfError: true);
-
-                     //        if (msiComponentKey.GetValueNames().Contains(userDataProductKeyId, StringComparer.OrdinalIgnoreCase))
-                     //        {
-                     //            _componentLoopCount++;
-                     //            appKey.InstallLocation = set_if_empty(appKey.InstallLocation, get_install_location_estimate(msiComponentKey.get_value(userDataProductKeyId)));
-                     //            if (!string.IsNullOrWhiteSpace(appKey.InstallLocation)) break;
-                     //            if (_componentLoopCount >= 10) break;
-                     //        }
-                     //    }
-                     //}

```

Comment:
Hey Rob,

Hmm yea.  If it was a work in progress i think it would be better suited in a branch that gets done when it gets done.  I'm okay with adding this back if you like you and simply adding a comment to the unused member variable to remove the warning.  Just let me know.

Thanks!

Reply:
Hey @ferventcoder,

Would u like me to add this code back for now?